src/Datadog.Trace.ClrProfiler.Managed/Integrations/Types/TypeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static class TypeExtensions
+    {
+        public static System.Type GetInstrumentedType(
```

Comment:
Thinking another method would be reasonable for getting interfaces as well. There are instances where we need to get generics from base types or from interfaces definitions.